### PR TITLE
unit tests changes

### PR DESCRIPTION
**Pull request by ashcas23**
Unit Test Changes: Added Unit tests regarding the post_detail, post_list, post_new, and post_edit pages. Created about four to five different tests in each to help ensure that the code works as intended and catches bugs early on.

### CODE REVIEW

**Comment by eclaros23** [comment]
Looks good to me!

File: blog/tests.py [hunk 1]
```diff
@@ -1,3 +1,176 @@
...
+        def test_post_loggedout(self):
+            """If user is logged out they wont be able to edit posts"""
+            self.client.logout()
+            author = User.objects.create (username = "hello_there101")
+            published_post = Post.objects.create(title = "Post1", text = "Post 1 data", author = author, published_date = now())
+            response = self.client.post(f"/post/{published_post.pk}/edit/")
+            self.assertEqual(response.status_code, 404) #making sure its sucessful
+            unpublished_post = Post.objects.create(title = "Post2", text = "Post 2 data", author = author)
+            response = self.client.post(f"/post/{unpublished_post.pk}/edit/")
+            self.assertEqual(response.status_code, 404)
```
Comment (dchukhin) [inline blog/tests.py:144]: After this test, could you also add a test for trying to edit a post that doesn't exist (for example, with a URL of `"/post/1000000/edit/"` or `"/post/!!!!!/edit/"`)?

File: blog/tests.py [hunk 2]
```diff
@@ -1,3 +1,176 @@
...
+            new_post = Post.objects.get(title = "new title", text = "new text")
+            self.assertRedirects(response,f'/post/{new_post.pk}/')
+
+        def test_post_not_logged_in(self):
+            """If user is not logged in they shouldnt be able to create new posts"""
+            self.client.logout() #since we are creating a new post we dont need an author or published posts
+            response = self.client.post("/post/new/")
+            self.assertEqual(response.status_code, 404)
+            response = self.client.post('/post/new/', {"title": "new title", "text": "new text"})
+            self.assertEqual(response.status_code, 404)
```
Comment (dchukhin) [inline blog/tests.py:183]: After this test, could you add a test that checks what happens if someone tries to create a post with the same title as another post that already exists?

**Review by dchukhin — COMMENTED** [review]
Good job with writing the tests! I left a couple of comments about some more tests that you could write.

**Review by dchukhin — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**